modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeVelocityLiferayTaglibReferenceMigrationCheck.java

```diff
@@ -0,0 +1,179 @@
+
+		for (int i = 0; i < lines.length; i++) {
+			String newLine = lines[i];
+
+			if (newLine.contains(_VELOCITY_LIFERAY_LANGUAGE_FORMAT)) {
+				String[] possibilities = {
+					_VELOCITY_LIFERAY_LANGUAGE_FORMAT +
+						_VELOCITY_LIFERAY_LANGUAGE_FORMAT_ARGUMENTS,
+					_VELOCITY_LIFERAY_LANGUAGE_FORMAT + CharPool.SPACE +
+						_VELOCITY_LIFERAY_LANGUAGE_FORMAT_ARGUMENTS
+				};
+
+				lines[i] = StringUtil.replace(
+					newLine, possibilities,
+					new String[] {
+						_FREEMARKER_LIFERAY_LANGUAGE_FORMAT,
+						_FREEMARKER_LIFERAY_LANGUAGE_FORMAT
+					});
+			}
+			else if (newLine.contains(_VELOCITY_LIFERAY_LANGUAGE)) {
+				String[] possibilities = {
+					_VELOCITY_LIFERAY_LANGUAGE + CharPool.OPEN_PARENTHESIS,
+					_VELOCITY_LIFERAY_LANGUAGE + CharPool.SPACE +
+						CharPool.OPEN_PARENTHESIS
+				};
+
+				newLine = StringUtil.replace(
+					newLine, possibilities,
+					new String[] {
+						_FREEMARKER_LIFERAY_LANGUAGE_KEY,
+						_FREEMARKER_LIFERAY_LANGUAGE_KEY
+					});
+
+				int languageKeyIndex = newLine.indexOf(
+					_FREEMARKER_LIFERAY_LANGUAGE_KEY);
+
+				String endLine = newLine.substring(languageKeyIndex);
+
+				String newEndLine = StringUtil.replaceFirst(
+					endLine, CharPool.CLOSE_PARENTHESIS,
+					CharPool.SPACE +
+						VelocityMigrationConstants.FREEMARKER_TAG_END);
+
+				lines[i] = StringUtil.replace(newLine, endLine, newEndLine);
+			}
+			else if (newLine.contains(_VELOCITY_LIFERAY_THEME_INCLUDE)) {
+				String[] possibilities = {
+					_VELOCITY_LIFERAY_THEME_INCLUDE +
+						CharPool.OPEN_PARENTHESIS + CharPool.DOLLAR,
+					StringBundler.concat(
+						_VELOCITY_LIFERAY_THEME_INCLUDE, StringPool.SPACE,
+						StringPool.OPEN_PARENTHESIS, StringPool.DOLLAR)
+				};
+
+				newLine = StringUtil.replace(
+					newLine, possibilities,
+					new String[] {
+						_FREEMARKER_LIFERAY_THEME_INCLUDE,
+						_FREEMARKER_LIFERAY_THEME_INCLUDE
+					});
+
+				lines[i] = StringUtil.replaceLast(
+					newLine, CharPool.CLOSE_PARENTHESIS,
+					CharPool.SPACE +
+						VelocityMigrationConstants.FREEMARKER_TAG_END);
+			}
+			else if (newLine.contains(_VELOCITY_LIFERAY_THEME_WRAP)) {
+				String[] possibilities = {
+					_VELOCITY_LIFERAY_THEME_WRAP + CharPool.OPEN_PARENTHESIS,
+					_VELOCITY_LIFERAY_THEME_WRAP + CharPool.SPACE +
+						CharPool.OPEN_PARENTHESIS
+				};
+
+				newLine = StringUtil.replace(
+					newLine, possibilities,
+					new String[] {
+						_FREEMARKER_LIFERAY_THEME_WRAP,
+						_FREEMARKER_LIFERAY_THEME_WRAP
+					});
+
+				int newStartIndex = newLine.indexOf(
+					_FREEMARKER_LIFERAY_THEME_WRAP);
+
+				int tabCount = StringUtil.count(
+					newLine.substring(0, newStartIndex), StringPool.TAB);
+
+				String tabulation = StringUtils.repeat(
+					StringPool.TAB, tabCount);
+
+				newLine = StringUtil.replace(newLine, ".vm", ".ftl");
+				newLine = StringUtil.replaceFirst(
+					newLine, StringPool.CLOSE_PARENTHESIS,
+					" />\n" + tabulation + "</@>", newStartIndex);
```

Comment:
@nicolas-sss, shouldn't we create constants to hold the strings ` />\n` and `</@>` as we've been doing? My idea is keep consistency :slightly_smiling_face: